resource/add_to_api.md

```diff
@@ -0,0 +1,216 @@
+Rather than create a new custom type for each field, we created a reusable `Rating` type that combines `measurement` & `upgrade`. It looks like this in the schema:
+
+```
+type Rating @cacheControl(maxAge: 90) {
+  measurement: Float
+  upgrade: Float
+}
+```
+If you encounter more nested data when adding fields, you can create your own custom types simply by defining them the same way that Rating is defined above.
+
+4. To see your new field in the API, save index.js then build & run the api:
+
+```
+yarn && yarn run build
+NRCAN_DB_CONNECTION_STRING="mongodb://localhost:27017" \
+	NRCAN_DB_NAME="energuide" \
+	NRCAN_COLLECTION_NAME="dwellings" \
+	NRCAN_ENGINE_API_KEY="your_apollo_engine_api_key" yarn start
+
+```
+
+Navigate to http://localhost:3000/. You should now be able to query airLeakage as follows:
+
+```
+query {
+  dwellings{
+    results {
+      evaluations {
+        airLeakage
+        }
+      }
+    }
+  }
+}
+```
+
+:tada:! If the field doesn't appear, be sure to check the spelling of the field name, and make sure you've run yarn build to rebuild the files.
+
+Next lets write some tests!
+
+## Adding a unit test for the new field
+
+### `api/src/schema/__tests__/schema.test.js`
+
+1. Add your new field to the `'Evaluation Type'` test. It should look like this:
+
+```
+describe('Evaluation Type', () => {
+  it('is defined', () => {
+    expect(typeMap).toHaveProperty('Evaluation')
+  })
+
+  it('has the expected fields', () => {
+    const Evaluation = typeMap.Evaluation
+    const fields = Object.keys(Evaluation.getFields())
+    expect(fields).toEqual([
+      ...
+      airLeakage,
+      ...
+    ])
+  })
+})
+
+```
+
+If you are adding a new `Dwelling` field, you would add it to the `Dwelling Type` test instead.
+
+2. In your console, run `yarn test`. All your tests should pass.
+
+## Adding an integration test for the new field
+
+### `api/test/queries.test.js`
+
+1. We need to test to make sure the field returns the data we expect. To do this, we need to pick a `houseID` and matching `airLeakage` values. You can get these values
+by looking at the data in the database, or you can start up the api as described earlier and run the following query:
+
+```
+query {
+  dwellings{
+    results {
+      houseId
+      evaluations {
+        airLeakage {
+            measurement
+            upgrade
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+Write down the first values for `houseId`, `measurement` & `upgrade` that the API returns. These will be your values for `testHouseId`, `testMeasurement` & `testUpgrade`
+in the following step.
+
+2. Above the `'retrieves all keys for ersRating data'` test, write the following test. Make sure to replace `testHouseId`, `testMeasurement` & `testUpgrade`
+with the values you wrote down in step 1.
+
+```
+it('retrieves all keys for airLeakage data', async () => {
```

Comment:
I understand it, but it makes me vaguely uncomfortable that we're advising them to structure the tests to conform to the untested codes output